protocol/implementations/js/src/types.ts

```diff
@@ -1,32 +1,76 @@
-export namespace TbDex {
-  export type Offering = {
-    placeholder: string;
-  }
-  
-  export type RFQ = {
-    test: string;
-  }
-  
-  export type Quote = {
-    placeholder: string;
-  }
-  
-  export type MessageTypes = {
-    rfq: RFQ,
-    quote: Quote
-  }
-  
-  export type MessageType<M extends keyof MessageTypes> = MessageTypes[M];
-  
-  export type Message<T extends keyof MessageTypes> = {
-    id: string;
-    contextId: string;
-    from: string;
-    to: string;
-    type: T;
-    body: MessageTypes[T];
-    dateCreated: string;
+export type TbDEXMessage = Metadata &
+  (
+    | {
+        type: "offering"
```

Comment:
When we use this structure rather than the generics in the old version, do we still get the ability to check at a type level, that the message body "is of type Quote"?

Reply:
ooh yea i liked the generics stuff moe put in. it's just that i didn't get the generic stuff when i used a generator to create these types for me. it would be nice if the schema just did that for me.

i've fixed it to manually re-incorporate generics.

Reply:
> used a generator to create these types for me

Oooh cool, so that goes from schema -> TS code?